source/languages/en/riak/dev/advanced/write-once.md

```diff
@@ -0,0 +1,137 @@
+Any attempt to create a bucket type with the `write_once` property set to `true`
+and with data types or the strong consistency property set will fail.
+
+The `write_once` property may not be set on the default bucket type, and may not
+be set on individual buckets.
+
+The `lww` and `allow_mult` properties are ignored, if the `write_once` property
+is set to `true`.
+
+The following example shows how to configure a bucket type with the `write_once`
+property:
+
+```
+prompt$ riak-admin bucket-type create my-bucket-type '{"props": {"write_once": true}}'
+my-bucket-type created
+
+prompt$ riak-admin bucket-type activate my-bucket-type
+my-bucket-type has been activated
+
+prompt$ riak-admin bucket-type status my-bucket-type
+my-bucket-type is active
+...
+write_once: true
+...
+```
+
+## Quorum
+
+The write once path supports the `W`, `PW`, and `DW` configuraton values.
+However, if `DW` is specified, then the value of `W` is taken to be the maximum
+of the `W` and `DW` values. For exmaple, for `n_val 3`, if `DW` is set to `all`,
+then `W` will be `3`.
+
+The write once path supports the `sloppy_quroum` property. If set to `false`,
+only primary nodes will be selected for calculation of write quorum nodes.
+
+## Runtime
+
+The "write once" path circumvents the normal coordinated put code path, and
+instead sends write requests directly to all vnodes (or vnode proxies) in the
+effective preference list for the write operation.
+
+In place of the `put_fsm` used in the normal path, we introduce a collection of
+new intermediate worker processes (implementing `gen_server` behavior). The role
+of these intermediate processes is to dispatch put requests to vnode(proxie)s in
+the preflist, and to aggregate replies. Unlike the `put_fsm`, the write once
+workers are long-lived for the lifecyle of the `riak_kv` application. They are
+therefore stateful, and store request state in a state-local dictionary.
+
+The relationship between the `riak_client`, write once workers, and vnode
+proxies is illustrated in the following diagram:
+
+<br>
+![Write Once](/images/write_once.png)
+<br>
+
+## Siblings
+
+As mentioned, entries in write once buckets are intended to be written only once
+-- users who are not abusing the semantics of the bucket type should not be
+updating or over-writing entries in buckets of this type. However, it is
+possible for users to mis-use the API, accidentally or otherwise, which might
+result in incomparable entries for the same key.
+
+In the case of siblings, write once buckets use an algorithm for resolving
+conflicts that has the appearance of random behavior from the user's point of
+view, but is in fact determinisitic. Effectively, Riak will choose the "least"
+entry among siblings, where the ordering relation is based on a deterministic
+SHA-1 hash of the object. This has the appearance to the user of "random write
+wins", even though the behavior is repeatable and determinisitic (assuming the
+same sibling inputs).
```

Comment:
Is the deterministic SHA-1 hash something the users can control/influence/impact? If so, then I agree that talking about it appearing to be random write is not as helpful. Instead, we could maybe suggest ways the user can impact the outcome.

If they can't impact the outcome, though, I would maybe rephrase to something like, 
"In the case of siblings, Write Once Buckets uses a deterministic algorithm based on SHA-1 hashes to resolve conflicts. While Riak uses the "least" entry among siblings, you do not have much control over which entry is least so the result may appear to be "random write wins"."